{% extends "base.html" %}

{% block content %}

<div class="basic-page">
    <div class="content">
        <h1>Questions and Answers</h1>
        <br>
        <p>
            <strong>1.  What is a Genome Wide Association Study (GWAS)?</strong>
        </p>
        <p>We have by now long understood the fact that the majority of genetic differences between individuals comes from variations in DNA. These differences are measured by genetic variants (Single-Nucleotide Polymorphisms) where humans are otherwise 99.9% identical to each other: it is the 0.1% by which we differ that makes us all unique. A methodological pipeline known as a Genome Wide Association Study (GWAS, pronounced 'Gee-WAS') is an unbiased and hypothesis-free approach which allows us to search the entire human genome in order to discover these variations, and how they are associated with a specific trait of interest (often interchangeably called a phenotype or outcome). Only in the case of 'Mendelian diseases' like Huntington's is there only one single associated gene. The majority of traits which we are interested in are complex and are the result of multiple genetic loci combining (referred to as 'polygenic traits'), where often hundreds and thousands of genetic variants have a small influence on a phenotype. First introduced in 2005, over 4,000 GWA studies have been conducted to date.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>2.	Why did we develop the GWAS Diversity Monitor?</strong>
        </p>
        <p>In our own GWAS, in the wider body of academic research, and through conference attendance, there is no getting away from the biases surrounding the data used for genomic research. For example, in our 2019 Communications Biology <a href="https://www.nature.com/articles/s42003-018-0261-x">review</a> of all GWAS until 2018, we estimated that 72% of all GWAS participants were recruited from just 3 countries (UK, US and Iceland). In the process of that work, we found that there was no systematic way for scientists, funders, editors, interests' groups and the general public to track changes in a systematic way. We hope that the GWAS Diversity Monitor will help to show where current research is lacking in terms of geography, disease, and ancestry groups and how this is varying over time. The GWAS Diversity Monitor dynamically outlines which diseases and traits have not been studied in which ancestry groups and provides easy and searchable links to each individual study and overall associations.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>3.  Can one download the data and graphs from the GWAS Diversity Monitor? Is there any replication material?</strong>
        </p>
        <p>Yes! We are advocates for Open Science and encourage you to download all of the graphs and underlying data which you need, with the hope that you will be able to easily integrate them into your own research. With regards to the data, full attribution is made to the NHGRI-EBI GWAS Catalog for raw data origination, and at this stage we would also like to point to the EMBL-EBI <a href="http://www.ebi.ac.uk/about/terms-of-use/">terms of use</a>. With regards to the code, this is made available under an MIT License (a permissive free software license) and can be found on the LCDS <a href="http://www.github.com/oxforddemsci/">GitHub page</a>.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>4.	Why is it important to diversify participants by ancestry?</strong>
        </p>
        <p>GWAS which utilize data from diverse populations will provide more accurately targeted therapeutic treatments to more of the world’s population, extend insights into the architecture of traits and uncover rare variants with significant effect sizes which replicate across ancestries. Isolated populations–due to bottleneck events, genetic drift, adaptation and selection–are of importance due to higher frequencies of rare variants which increase the power to detect associations with clinically important phenotypes.

        The diagnoses among and the design of treatments for European ancestry populations are being blindly applied on other groups such as African and African-Americans.  For instance, a recent study published in <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)31120-1">Cell</a> contained data on people from rural Uganda and identified 10 new links between diseases and genes which had been previously undetected. They found that 22 percent of people have a gene that causes the blood disorder thalassemia. This 'disorder' has in fact become common in the African population since it is protective against severe malaria. The genetic variant is linked to glycated haemoglobin which is currently used (based on discoveries from European populations) to diagnose diabetes, leading to incorrect diagnoses in Africans.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>5.	Why do we expect to find more genetic insights when looking beyond European populations?</strong>
        </p>
        <p>  Due to an increased genetic diversity in non-European ancestry populations, considerably more insight into disease and treatments will be reaped from studying just tens of thousands of participants that wouldn’t be found if we studied even hundreds of thousands of Europeans. Increasing the number of different combinations of DNA which are studied can have huge scientific benefits, and recent genetic sequencing of Khoi-San bushmen showed that even two people from adjacent Southern African villages were as different from one another as any two European individuals.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>6.	What is the Experimental Factor Ontology?</strong>
        </p>
        <p>  The Experimental Factor Ontology (EFO) provides a systematic description of many experimental variables available in EBI databases, and for projects such as the NHGRI-EBI GWAS Catalog. Please see the EFO page on the EMBL-EBI Ontology Lookup Service <a href="https://github.com/EBISPOT/efo/tree/master">GitHub</a> for more information.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>7.	How are the ancestry terms defined?</strong>
        </p>
        <p> Please see the supplementary material from our accompanying article for a full mapping from the ‘Broad’ ancestral category to our engineered ‘Broader’ field.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>8.	Are race and ancestry the same thing?</strong>
        </p>
        <p>Race and Ancestry should not be used as interchangeable terms. As described in detail elsewhere, one common misconception is that the ancestry (population structure) used in analyses equate to racial or ethnic differences: a statement that is patently incorrect. The terms 'ancestry' and 'race' in human genetics are not interchangeable.

        Genetic variation is different from the social, cultural, and political meanings ascribed to different human groups. Race is not a biological category since genetic variation is traced to geographical locations and does not map into the perpetually changing and socially and politically defined racial or ethnic groups. Populations are the product of repeated mixtures over tens of thousands of years. The concentration of genetic alleles in some groups is thus related to where they have descended from and has nothing to do with the social category of race. For example, the ancestral category of ‘African-American’ is highly diverse or what is called 'admixed' and can be traced in relation migration and assortative mating patterns.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>9.	Why do some ancestry groups have more associations and a higher allele frequency than others?</strong>
        </p>
        <p> Most evidence places the origins of Homo sapiens in Africa. This is since sub-Saharan Africa is where patterns of DNA sequence variation are the greatest, and populations with the greatest genetic variation are assumed to be the oldest. Current knowledge places the cradle of the human species in what is now modern Namibia and Angola in Southern Africa. This is attributed to the fact that when humans migrated to new regions, they took progressively smaller amounts of genetic variation in the gene pool with them. Each new population is younger than its original source and thus has less time to accumulate new mutations.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>10.	Why can’t the results from GWAS from European ancestry populations be applied to other ancestry groups?</strong>
        </p>
        <p> Polygenic scores derived from one population are not reliable when applied to others. For example, the results for BMI (body mass index) derived from a GWAS from a European population only explains 3% of the variance for individuals of African ancestry compared to 13% of those with European ancestry. With the 95% confidence intervals overlapping zero, the genetic score for BMI based on European ancestry does not significantly explain the variance of BMI among individuals with African ancestry at all.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>11.	Why is it important to look beyond ancestry and to historical, geographical and other environmental differences? </strong>
        </p>
        <p> Although ancestry has often been the focus of the diversity discussion in genetics, as we have shown in previous <a href="https://www.nature.com/articles/s41562-017-0195-1">research</a>, findings also largely differ by historical period, birth cohort and the geographical location and context of individuals. In future work to expand this site we hope to examine diversity beyond ancestry to encompass differences related to sex, socio-economic status, geography, birth cohort, historical period and other factors that impact this research.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>11.	Why is it important to look beyond ancestry and to historical, geographical and other environmental differences? </strong>
        </p>
        <p> Although ancestry has often been the focus of the diversity discussion in genetics, as we have shown in previous research, findings also largely differ by historical period and birth cohort and the geographical location and context of individuals. In future work to expand this site we hope to examine diversity beyond ancestry to encompass differences related to sex, socio-economic status, geography, birth cohort, historical period and other factors that impact this research.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>12.	Why is now the time to consider European bias in genetic data and research? </strong>
        </p>
        <p> Given the full release of the UK Biobank and increased reliance on large direct-to-consumer data, we predict that diversity in GWAS ancestry may decrease even further given that 94.23% of the 488,377 genotyped UK Biobank participants are in the white ethnic group and 23andMe has a sample with 77% European ancestry.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>13.	What do you mean by quasi-real time monitoring? Why do some of the numbers mentioned in the article differ from the online dashboard? </strong>
        </p>
        <p> Please note that given the 'live', on-going nature of the GWAS Diversity Monitor, the figures will update and may not concur exactly with those in the published article. Quasi real-time refers to the fact that we check the EMBL-EBI every morning for new.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>14.	Why do some of the numbers of total GWAS diversity differ across the differ figures? </strong>
        </p>
        <p> The 'Total GWAS participants diversity' in the upper left corner represents total, unfiltered estimates of GWAS participant diversity (dropping all rows of the Catalog which have at least one ancestry which is not reported). All of the other figures filter or aggregate in some way with regards to Stage, Parent Term, Ancestry, or for some specific point in time.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>15.	What are the strengths of the GWAS Diversity Monitor?</strong>
        </p>
        <p>Our strengths are:
          <ul>
            <li>A visual, systematic and regular analysis of diversity in all GWAS to date.</li>
            <li>A tool for researchers, funders, editors, interest groups and the general public</li>
            <li>A detailed longitudinal analysis over time.</li>
            <li>A breakdown of ancestry across discovery and replication phases.</li>
            <li>Identification of the main countries used for participant recruitment.</li>
            <li>Visualization of the associations discovered by ancestry group.</li>
          </ul>
        </p>
    </div>
    <div class="content">
    <p>
        <strong>What are the limitations of the GWAS Diversity Monitor?</strong>
    </p>
    <p>We were limited by:
      <ul>
        <li>A lack of specificity by authors in describing their study participants.</li>
        <li>The ancestry groups that geneticists define: With increasing admixture and migration, categorization increasingly demands constant re-evaluation.</li>
        <li>The lack of Digital Object Identifiers which identify genetic datasets, which would allow for a richer data curation process and subsequent analysis.</li>
      </ul>
    </p>
</div>
</div>

{% endblock %}
