{% extends "base.html" %}

{% block content %}

<div class="basic-page">
    <div class="content">
        <h1>Questions and Answers</h1>
        <br>
        <p>
            <strong>1.  What is a Genome Wide Association Study (GWAS)?</strong>
        </p>
        <p>Since around the 1970s, we know that the majority of genetic differences between individuals comes from variations in DNA sequences. These differences are measured by genetic variants (Single-Nucleotide Polymorphisms). Humans are otherwise 99.9% identical to each other and it is this 0.1% by which we differ that makes us all unique. To study these differences, researchers developed a methodological approach known as a Genome Wide Association Study (GWAS, pronounced 'Gee-WAS'), which is a hypothesis-free method that searches across the entire genome to discover how these SNP variations are associated with a specific trait of interest. First introduced in 2005, over 4,000 GWA studies have been conducted to date. It is virtually never the case that it is ‘one’ SNP or gene that is associated with a trait, but often many. Only in the case of 'Mendelian diseases' like Huntington's is there only one single associated gene. The majority of traits which we are interested in are complex and are the result of multiple genetic loci (referred to as polygenic).
        </p>
    </div>

    <div class="content">
        <p>
            <strong>2.  Why did we develop the GWAS Diversity Monitor?</strong>
        </p>
        <p>There is increasing recognition in the field that GWAS and genetics research uses data from highly selective groups of people that do not equally represent the inhabitants of people from that country or the world.  For example, in our 2019 Communications Biology <a href="https://www.nature.com/articles/s42003-018-0261-x" target="_blank">review</a> of all GWAS until the end of 2018, we estimated that 72% of all GWAS participants were recruited from just 3 countries (UK, US and Iceland) (1). We also found a strong underrepresentation of people from non-European ancestry groups. When non-European groups were included, such as from Asian or African populations, it was generally only to replicate original discovery results from European groups and not to make fundamentally new discoveries within that group. This also made us curious as to whether there was selection by particular diseases that were studied, which is why we developed the ‘heatmap’ plots you see on the online GWAS Diversity Monitor.

        We also realized that there was no systematic way for scientists, funders, editors, interest groups and the general public to understand if diversity and representation was getting better or worse. Although everyone agrees it is important, there wasn’t any way to track progress. For this reason, we developed the GWAS Diversity Monitor to track where current research is lacking in terms of geography, disease, and ancestry groups, and how this varies over time. The GWAS Diversity Monitor dynamically outlines which diseases and traits have (and have not) been studied across which ancestry groups over time and provides easy and searchable links to each individual study searchable by individual trait.

        </p>
    </div>

    <div class="content">
        <p>
            <strong>3.  Where does the data driving the GWAS Diversity Monitor come from?</strong>
        </p>
        <p>Our GWAS Diversity Monitor is driven by data taken from the NHGRI-EBI GWAS Catalog, with detailed information on the eligibility criteria available elsewhere (2-4) and the <a href="https://www.ebi.ac.uk/gwas/" target="_blank">NHGRI-EBI Catalog</a> itself. For inclusion in the Catalog, studies are identified via an automated PubMed literature search and then examined against matching the inclusion criteria. Trained curators then include the information in the Catalog using a pre-defined protocol.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>4.  Can I download the data and graphs from the GWAS Diversity Monitor? Is there any replication material?</strong>
        </p>
        <p>Yes! We are advocates for Open Science and encourage you to download all of the graphs and underlying data you need, with the hope that you will be able to easily integrate it into your own research. With respect to the data, we give full attribution to the <a href="https://www.ebi.ac.uk/gwas/" target="_blank">NHGRI-EBI Catalog</a> which is where we draw the original raw data and point to their EMBL-EBI <a href="https://www.ebi.ac.uk/about/terms-of-use/" target="_blank">terms of use</a>. With regards to the code, we have made this available under an MIT License (a permissive free software license) and it can be found on the Leverhulme Centre for Demographic Science (LCDS) <a href="https://github.com/OxfordDemSci/gwasdiversitymonitor" target="_blank">GitHub</a> page. We would love to hear from you, so please do let us know if and how this has been integrated into your work.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>5.  What does the category ‘Parent term’ mean?</strong>
        </p>
        <p>Our online dashboard divides the traits that are studied across GWAS by the standard Experimental Factor Ontology (EFO), with the most aggregated or highest-level categories often referred to as 'Parent terms'. This provides a systematic description of many experimental variables available in EBI databases, and for projects such as the NHGRI-EBI GWAS Catalog. Please see the EFO page on the EMBL-EBI Ontology Lookup Service <a href="https://github.com/EBISPOT/efo/tree/master" target="_blank">GitHub</a> for more information.
        </p>
    </div>


    <div class="content">
        <p>
            <strong>6.  How are the groups divided by 'Ancestry'?</strong>
        </p>
        <p>The GWAS Diversity Monitor is divided into 6 broad ancestry groups, based on a framework developed by Morales et al. (2018)(4), which creates a standard for ancestry classifications in GWAS research. You can find a detailed description of these groups and various examples in the Supplementary Material linked to our original article in Table S1. Although we recognize that it is suboptimal genetically to categorize groups into such broad categories, this type of assignment reduces complexity and enables visualizations and categorisation. Furthermore, since most non-European ancestry groups remain underrepresented, disaggregating to a more detailed level is often not useful. Our hope is that as diversity increases, disaggregated groups will more frequently represented in the future.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>7. Are race and ancestry the same thing?</strong>
        </p>
        <p>Race and ancestry are not interchangeable terms. As described in detail elsewhere (5), one common misconception is that the term ancestry used in population genetics equates to racial or ethnic differences: a statement that is patently incorrect. Genetic variation in populations (ancestry) is different from the social, cultural, and political meanings ascribed to different human groups. Race is not a biological category since genetic variation is traced to geographical locations and does not map into the perpetually changing and socially and politically defined racial or ethnic groups. Populations are the product of repeated mixtures over tens of thousands of years. The concentration of genetic alleles in some groups is thus related to where they have descended from and has nothing to do with the socially constructed category of race. For example, the ancestral category of 'African-American' is highly diverse or what is called 'admixed' and can be traced in relation to migration and assortative mating patterns.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>8.  Why is there a difference in the donut charts of ‘participants by ancestry’ and ‘count of all associations discovered’? Why do ancestry groups have different allele frequencies?</strong>
        </p>
        <p>Most evidence places the origins of Homo sapiens in Africa. Sub-Saharan Africa is where patterns of DNA sequence variation are the greatest, and populations with the greatest genetic variation are assumed to be the oldest. Current knowledge places the cradle of the human species in what is now modern Namibia and Angola in Southern Africa. This is attributed to the fact that when humans migrated to new regions, they took progressively smaller amounts of genetic variation in the gene pool with them. Each new population is younger than its original source and thus has less time to accumulate new mutations, with European ancestry groups having the least variation. Sequencing of Khoi-San bushmen showed that even two people from adjacent villages were as different from one another as any two European or non-African ancestry individuals.(6-7)
        </p>
    </div>

    <div class="content">
        <p>
            <strong>9.  Why can’t the results from GWAS derived from European ancestry populations simply be applied to other ancestry groups?</strong>
        </p>
        <p>Polygenic scores derived from one population are not reliable when applied to others (8). For example, the results for BMI (body mass index) derived from a GWAS from a European population only explains 3% of the variance for individuals of African ancestry compared to 13% of those with European ancestry. With the 95% confidence intervals overlapping zero, the genetic score for BMI based on European ancestry does not significantly explain the variance of BMI among individuals with African ancestry at all.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>10. Why is it important to look at diversity beyond ancestry and towards historical, geographical, demographic differences?</strong>
        </p>
        <p>Although ancestry has often been the focus of the diversity discussion in genetics, as we have shown in previous <a href="https://www.nature.com/articles/s41562-017-0195-1" target="_blank">research</a>, genetic findings also largely differ by historical period, birth cohort and the geographical location, demographics and socioeconomic context of individuals (9). In future work to expand this site we hope to examine diversity beyond ancestry to encompass differences related to sex, socio-economic status, geography, birth cohort, historical period and other factors that impact this research.
        </p>
    </div>


    <div class="content">
        <p>
            <strong>11. Why is it important to get more diverse populations in genetics and genomics research??</strong>
        </p>
        <p>GWA studies that draw from data from diverse populations will provide more accurately targeted therapeutic treatments to more of the world’s population, extend insights into the genetic architecture of traits and very likely make new genetic discoveries. Diagnoses, treatments and interventions derived from European ancestry populations cannot be easily applied to other groups and if they are, they could even cause damage. For instance, a recent study published in <a href="https://www.cell.com/cell/fulltext/S0092-8674(19)31120-1" target="_blank">Cell</a> contained data on people from rural Uganda and identified 10 new links between diseases and genes which had been previously undetected (10). They found that 22 percent of people in the sample have a gene that causes the blood disorder thalassemia. This supposed ‘disorder’ has in fact become common in the African population since it is protective against severe malaria. The genetic variant is linked to glycated haemoglobin which is currently used (based on discoveries from European populations) to diagnose diabetes, which in turn would lead to incorrect diagnoses of diabetes in this African population. We also anticipate more discoveries from non-European samples. Due to an increased genetic diversity in non-European ancestry populations, considerably more insight into disease and treatments will be reaped from studying just tens of thousands of participants that wouldn’t be found if we studied even hundreds of thousands of Europeans. Increasing the number of different combinations of DNA which are studied can have huge scientific benefits.

        </p>
    </div>

    <div class="content">
        <p>
            <strong>12. Why should we highlight the lack of diversity?</strong>
        </p>
        <p>Given the full release of the UK Biobank and increased reliance on large direct-to-consumer data, we predict that diversity in GWAS ancestry may decrease even further given that 94.23% of the 488,377 genotyped UK Biobank participants are in the white ethnic group and 23andMe has a sample with 77% European ancestry. This acknowledgement can give further impetus to large scale data collection that is currently taking place in Africa, large projects such as NIH's <a href="https://www.ebi.ac.uk/gwas/" target="_blank">All of Us</a>, or new initiatives such as the UK's plans for a new 5 million person cohort.
        </p>
    </div>


    <div class="content">
        <p>
            <strong>13. What do you mean by quasi-real time monitoring? Why do some of the numbers mentioned in the article differ from the online dashboard?</strong>
        </p>
        <p>Note that given the ‘live’ and on-going dynamic nature of the GWAS Diversity Monitor, the figures update daily and may not concur exactly with those in the published article. Quasi real-time refers to the fact that we check the EMBL-EBI each morning and automatically update the monitor
        </p>
    </div>

    <div class="content">
        <p>
            <strong>14. Why do some of the numbers of total GWAS diversity differ across the different figures?</strong>
        </p>
        <p>The ‘Total GWAS participants diversity’ in the upper left corner represents total, unfiltered estimates of GWAS participant diversity (dropping all rows of the Catalog which have at least one ancestry which is not reported). All of the other figures filter or aggregate in some way with respect to Stage, Parent Term, Ancestry, or for some specific point in time.
        </p>
    </div>

    <div class="content">
        <p>
            <strong>15. What are the strengths of the GWAS Diversity Monitor?</strong>
        </p>
        <p>Our strengths are:
          <ul>
            <li>A visual, systematic and regular analysis of diversity in all GWAS to date.</li>
            <li>A tool for researchers, funders, editors, interest groups and the general public</li>
            <li>A detailed longitudinal analysis over time.</li>
            <li>A breakdown of ancestry across discovery and replication phases.</li>
            <li>A breakdown of ancestry across traits studied to identified diseases and traits that are understudied in certain groups.</li>
            <li>Identification of the main countries used for participant recruitment.</li>
            <li>Visualization of the associations discovered by ancestry group.</li>
          </ul>
        </p>
    </div>
    <div class="content">
    <p>
        <strong>16. What are the limitations of the GWAS Diversity Monitor?</strong>
    </p>
    <p>We were limited by:
      <ul>
        <li>A lack of specificity by authors in describing their study participants.</li>
        <li>The ancestry groups that geneticists define: With increasing admixture and migration, categorization increasingly demands constant re-evaluation.</li>
        <li>The lack of Digital Object Identifiers which identify genetic datasets, which would allow for a richer data curation process and subsequent analysis.</li>
      </ul>
    </p>

    <div class="content">
    <p>
        <strong>References cited</strong>
    </p>
    <p>
      <ol><li> Mills, M. C. & Rahal, C. A Scientometric Review of Genome-Wide Association Studies. Commun. Biol. 2, (2019).
        <li> Welter, D. et al. The NHGRI GWAS Catalog, a curated resource of SNP-trait associations. Nucleic Acids Res. 42, D1001–D1006 (2014).</li>
        <li> MacArthur, J. et al. The new NHGRI-EBI Catalog of published genome-wide association studies (GWAS Catalog). Nucleic Acids Res. 45, D896–D901 (2017).</li>
        <li> Morales, J. et al. A standardized framework for representation of ancestry data in genomics studies, with application to the NHGRI-EBI GWAS Catalog. Genome Biol. 19, 21 (2018).</li>
        <li> Mills, M. C., Barban, N. & Tropf, F. C. An Introduction to Statistical Genetic Data Analysis. (The MIT Press, 2020).</li>
        <li> Schuster, S. C. et al. Complete Khoisan and Bantu genomes from southern Africa. Nature 463, 943–947 (2010).</li>
        <li> Lachance, J. et al. Evolutionary History and Adaptation from High-Coverage Whole-Genome Sequences of Diverse African Hunter-Gatherers. Cell 150, 457–469 (2012).</li>
        <li> Martin AR, Gignoux CR, Walters RK, Wojcik GL, Neale BM, et al. Human Demographic History Impacts Genetic Risk Prediction across Diverse Populations. Am. J. Hum. Genet. 100(4):635–49 (2017).</li>
        <li> Tropf, F. C. et al. Hidden heritability due to heterogeneity across seven populations. Nat. Hum. Behav. 1, 757–765 (2017).</li>
        <li> Gurdasani, D. et al. Uganda Genome Resource Enables Insights into Population History and Genomic Discovery in Africa. Cell 179, 984-1002.e36 (2019).</li>
      </ol>
    </p>
</div>
</div>

{% endblock %}
